src/hotspot/share/classfile/modules.cpp

```diff
@@ -724,7 +724,12 @@ const char* Modules::get_numbered_property_as_sorted_string(const char* property
   }
 
   const char* result = (const char*)os::strdup(st.as_string()); // Example: "java.base,java.compiler"
-  return strcmp(result, "") != 0 ? result : nullptr;
+  if (strcmp(result, "") != 0) {
+    return result;
+  } else {
+    os::free((void*)result);
+    return nullptr;
+  }
```

Comment:
I think it would be slightly better to avoid the strdup in the empty string case:
```
const char* str = st.as_string();
if (str[0] != 0) {
  return os::strdup(str);
} else {
  return nullptr;
}
```

Reply:
Or slightly cleaner:
```
return (st.size() > 0)  ? os::strdup(st.as_string()) : nullptr;
```

Reply:
Thanks David, avoid strdup in the empty string case also make ASAN silently, thanks.